sweep/test_utils.go

```diff
@@ -155,12 +155,26 @@ func (m *MockNotifier) RegisterBlockEpochNtfn(
 	log.Tracef("Mock block ntfn registered")
 
 	m.mutex.Lock()
-	epochChan := make(chan *chainntnfs.BlockEpoch, 0)
-	bestHeight := int32(0)
-	if bestBlock != nil {
-		bestHeight = bestBlock.Height
+	epochChan := make(chan *chainntnfs.BlockEpoch)
+
+	// The real notifier returns a notification with the current block hash
+	// and height immediately if no best block hash or height is specified
+	// in the request. We want to emulate this behaviour as well for the
+	// mock.
+	switch {
+	case bestBlock == nil:
+		// Send out the notification immediately but do it in a
+		// goroutine to not block execution of this method.
+		go func() {
+			epochChan <- &chainntnfs.BlockEpoch{
+				Hash:   mockChainHash,
+				Height: mockChainHeight,
+			}
+		}()
+		m.epochChan[epochChan] = mockChainHeight
```

Comment:
@joostjager How would buffering the channel with 1 and sending the block immediately lead to surprises to future devs?  This is how we usually add elements to a channel before returning afaik, and I would prefer that approach to a goroutine  also here.

Reply:
Buffering with 1 is ok. But in the previous iteration it would make only a buffered channel in case of a `bestBlock` being present. Otherwise it would be an unbuffered channel. That was what I meant by surprise - it is kind of unclear what kind of channel you get and how it behaves in the test. If you can always buffer it with one element (but didn't check if that works with the test), I also prefer that over the goroutine.

Question about the code as it currently is, is whether there can be a race condition with the goroutine and a line in the test also passing in a block. This would make the ordering non-deterministic.

Reply:
Ah, I see.  I think we can just buffer it with 1 always then, or just keep the goroutine. I'm fine with either.

Reply:
I was able to change it back to buffer size 1 and remove the goroutine without any impact. Apparently the mock notifier hasn't really been used for a test before or didn't expect a specific buffer size.